WalletWasabi/Hwi2/HwiClient.cs

```diff
@@ -0,0 +1,197 @@
+				command: HwiCommands.Wipe,
+				commandArguments: null,
+				cancel).ConfigureAwait(false);
+		}
+
+		public async Task SetupAsync(HardwareWalletVendors deviceType, string devicePath, CancellationToken cancel)
+		{
+			await SendCommandAsync(
+				options: new[] { HwiOption.DevicePath(devicePath), HwiOption.DeviceType(deviceType), HwiOption.Interactive },
+				command: HwiCommands.Setup,
+				commandArguments: null,
+				cancel).ConfigureAwait(false);
+		}
+
+		public async Task RestoreAsync(HardwareWalletVendors deviceType, string devicePath, CancellationToken cancel)
+		{
+			await SendCommandAsync(
+				options: new[] { HwiOption.DevicePath(devicePath), HwiOption.DeviceType(deviceType), HwiOption.Interactive },
+				command: HwiCommands.Restore,
+				commandArguments: null,
+				cancel).ConfigureAwait(false);
+		}
+
+		public async Task BackupAsync(HardwareWalletVendors deviceType, string devicePath, CancellationToken cancel)
+		{
+			if (deviceType == HardwareWalletVendors.Trezor)
+			{
+				// HWI would throw the same, don't need the roundtrip.
+				throw new HwiException(HwiErrorCode.UnavailableAction, "The Trezor does not support creating a backup via software");
+			}
+
+			await SendCommandAsync(
+				options: new[] { HwiOption.DevicePath(devicePath), HwiOption.DeviceType(deviceType), HwiOption.Interactive },
+				command: HwiCommands.Backup,
+				commandArguments: null,
+				cancel).ConfigureAwait(false);
+		}
+
+		public async Task<Version> GetVersionAsync(CancellationToken cancel)
+		{
+			string responseString = await SendCommandAsync(options: new[] { HwiOption.Version }, command: null, commandArguments: null, cancel).ConfigureAwait(false);
+
+			var version = HwiParser.ParseVersion(responseString);
+			return version;
+		}
+
+		public async Task<string> GetHelpAsync(CancellationToken cancel)
+		{
+			string responseString = await SendCommandAsync(options: new[] { HwiOption.Help }, command: null, commandArguments: null, cancel).ConfigureAwait(false);
+
+			return responseString;
+		}
+
+		public async Task<IEnumerable<HwiEnumerateEntry>> EnumerateAsync(CancellationToken cancel)
+		{
+			string responseString = await SendCommandAsync(options: null, command: HwiCommands.Enumerate, commandArguments: null, cancel).ConfigureAwait(false);
+			IEnumerable<HwiEnumerateEntry> response = HwiParser.ParseHwiEnumerateResponse(responseString);
+
+			return response;
+		}
+
+		public async Task<string> SetupAsync(CancellationToken cancel)
+		{
+			string responseString = await SendCommandAsync(options: null, command: HwiCommands.Setup, commandArguments: null, cancel).ConfigureAwait(false);
+			return responseString;
+		}
+
+		public async Task<string> DisplayAddressAsync(CancellationToken cancel)
+		{
+			string responseString = await SendCommandAsync(options: null, command: HwiCommands.DisplayAddress, commandArguments: null, cancel).ConfigureAwait(false);
+			return responseString;
```

Comment:
Most of these could have the return inlined instead of saving the response to a variable